### PR TITLE
Added PCA + Tests

### PR DESCRIPTION
**Pull request by raviselker**

### CODE REVIEW

File: R/pca.b.R [hunk 1]
```diff
@@ -0,0 +1,339 @@
+
+# This file is a generated template, your changes will not be overwritten
```
Comment (jonathon-love) [inline R/pca.b.R]: you can get rid of this

File: R/pca.b.R [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+
+pcaClass <- R6::R6Class(
+    "pcaClass",
+    inherit = pcaBase,
+    private = list(
+        #### Global variables ----
```
Comment (jonathon-love) [inline R/pca.b.R]: so these technically wouldn't be global, they're 'member variables'

File: R/pca.b.R [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+
+            nFactors <- private$.nFactors(data)
+
+            suppressWarnings({
+
+                r <- psych::principal(data, nfactors = nFactors, rotate = self$options$rotation)
+
+                devnull <- '/dev/null'
+                if (Sys.info()['sysname'] == 'Windows')
+                    devnull <- 'NUL'
```
Comment (jonathon-love) [inline R/pca.b.R:49]: have you checked this on windows? because i haven't.

File: R/pca.b.R [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+
+            for (var in vars) {
+
+                row <- list()
+                for (j in 1:nFactors) {
+
+                    index <- which(colNames %in% paste0("C",j))
+
+                    l <- loadings[jmvcore::toB64(var), index]
+                    hide <- self$options$hideLoadings
```
Comment (jonathon-love) [inline R/pca.b.R]: you could shift this assignment above the loop. you could put it with `vars <- self$options$vars`. keep all these guys together.

File: R/pca.b.R [hunk 5]
```diff
@@ -0,0 +1,339 @@
...
+            eigen <- eigen(corMatrix)$values
+            # eigenValuesEFA <- psych::fa(rx, fm="minres", warnings=FALSE)$values
+
+            nSub <- dim(data)[1]
+            nVar <- dim(data)[2]
+
+            # Simulate eigen values
+            simEigenList <- parallel::mclapply(1:nIter, function(XX) {
+
+                # bad <- TRUE
```
Comment (jonathon-love) [inline R/pca.b.R:309]: did you want to tidy up this commented out stuff? remove it?

File: jamovi/pca.a.yaml [hunk 6]
```diff
@@ -0,0 +1,107 @@
+---
+name:  pca
+title: Principal Component Analysis
+menuGroup: Factor
+version: '1.0.0'
+jas: '1.0'
```
Comment (jonathon-love) [inline jamovi/pca.a.yaml:6]: do you want to add a `usage` entry? take a look at a .a.yaml file from a different analysis.

(but put spaces around the ` = ` signs, i haven't done that, but i'm going to change it)

File: R/pca.b.R (hunk 3 above)
Comment (raviselker) [reply R/pca.b.R:49]: I don't have a windows build so it's difficult for me to check on windows

File: R/pca.b.R (hunk 5 above)
Comment (raviselker) [reply R/pca.b.R:309]: I might need this code in the future so I'd rather keep it in for now

File: jamovi/pca.a.yaml (hunk 6 above)
Comment (raviselker) [reply jamovi/pca.a.yaml:6]: Added

**Comment by jonathon-love** [comment]
i've merged this, but i squashed your commits 1 and 3 together, so you'll have to do the `git reset HEAD~3 --hard` thing before rebasing.

**Pull request closed without merging**